service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderGenerationService.java

```diff
@@ -233,42 +235,53 @@ private DocmosisRepresentative getApplicantDetails(String applicantName, Solicit
         return applicantDetails.build();
     }
 
-    private List<DocmosisDirection> getGroupedCMODirections(CaseManagementOrder caseManagementOrder) {
-        return ofNullable(caseManagementOrder.getDirections()).map(directions -> {
-                ImmutableList.Builder<DocmosisDirection> formattedDirections = ImmutableList.builder();
-
-                int directionNumber = 1;
-                for (Element<Direction> element : directions) {
-                    Direction direction = element.getValue();
-                    if (direction.getParentsAndRespondentsAssignee() != null) {
-                        formattedDirections.add(DocmosisDirection.builder()
-                            .header("For " + direction.getParentsAndRespondentsAssignee().getLabel())
-                            .assignee(direction.getAssignee())
-                            .title(formatTitle(directionNumber++, direction))
-                            .body(direction.getDirectionText())
-                            .build());
-                    }
-
-                    if (direction.getOtherPartiesAssignee() != null) {
-                        formattedDirections.add(DocmosisDirection.builder()
-                            .header("For " + direction.getOtherPartiesAssignee().getLabel())
-                            .assignee(direction.getAssignee())
-                            .title(formatTitle(directionNumber++, direction))
-                            .body(direction.getDirectionText())
-                            .build());
-
-                    } else {
-                        formattedDirections.add(DocmosisDirection.builder()
-                            .assignee(direction.getAssignee())
-                            .title(formatTitle(directionNumber++, direction))
-                            .body(direction.getDirectionText())
-                            .build());
-                    }
+    private List<DocmosisDirection> getGroupedCMODirections(List<Element<Direction>> directions) {
+        return ofNullable(directions).map(this::buildDocmosisDirections).orElse(ImmutableList.of());
+    }
+
+    private ImmutableList<DocmosisDirection> buildDocmosisDirections(List<Element<Direction>> directions) {
+        ImmutableList.Builder<DocmosisDirection> formattedDirections = ImmutableList.builder();
+
+        int directionNumber = 2;
+        ParentsAndRespondentsDirectionAssignee respondentHeader = null;
+        OtherPartiesDirectionAssignee otherHeader = null;
+
+        for (Element<Direction> element : directions) {
+            Direction direction = element.getValue();
+
+            if (direction.getParentsAndRespondentsAssignee() != null) {
+                DocmosisDirection.Builder builder = buildBaseDirection(direction, directionNumber++);
```

Comment:
If you want this could be extracted to outside the `if`, and the add also could be extracted to the end of the `if`

Reply:
will update because that's a good little change